Bottom: a160a764d0290a1741b1c900a3dd0139368d391e
Top:    97f5844150e4f68606a52b7896e5578fb7ab7e5b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-23 17:41:30 +0000

Remove Draw() call from ResetFound*() Let the caller do it

This allows multiple changes to be grouped into one repaint operation.
Use this where appropriate, such as the Netlist lookup functions.


---

diff --git a/src/action.c b/src/action.c
index c4c28c0..d8e631f 100644
--- a/src/action.c
+++ b/src/action.c
@@ -2365,17 +2365,26 @@ ActionConnection (int argc, char **argv, int x, int y)
 
 	case F_ResetLinesAndPolygons:
 	  if (ResetFoundLinesAndPolygons (true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	  break;
 
 	case F_ResetPinsViasAndPads:
 	  if (ResetFoundPinsViasAndPads (true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	  break;
 
 	case F_Reset:
 	  if (ResetConnections (true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	  break;
 	}
       RestoreCrosshair (true);
@@ -2789,7 +2798,10 @@ ActionDisplay (int argc, char **argv, int childX, int childY)
 	  if (TEST_FLAG (AUTODRCFLAG, PCB) && Settings.Mode == LINE_MODE)
 	    {
 	      if (ResetConnections (true))
-		IncrementUndoSerialNumber ();
+		{
+		  IncrementUndoSerialNumber ();
+		  Draw ();
+		}
 	      if (Crosshair.AttachedLine.State != STATE_FIRST)
 		LookupConnection (Crosshair.AttachedLine.Point1.X,
 				  Crosshair.AttachedLine.Point1.Y, true, 1,
diff --git a/src/find.c b/src/find.c
index 345e670..748431e 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3486,11 +3486,7 @@ ResetFoundPinsViasAndPads (bool AndDraw)
   }
   END_LOOP;
   if (change)
-    {
-      SetChangedFlag (true);
-      if (AndDraw)
-        Draw ();
-    }
+    SetChangedFlag (true);
   return change;
 }
 
@@ -3555,11 +3551,7 @@ ResetFoundLinesAndPolygons (bool AndDraw)
   }
   ENDALL_LOOP;
   if (change)
-    {
-      SetChangedFlag (true);
-      if (AndDraw)
-        Draw ();
-    }
+    SetChangedFlag (true);
   return change;
 }
 
@@ -3922,7 +3914,10 @@ DRCAll (void)
   TheFlag = FOUNDFLAG | DRCFLAG | SELECTEDFLAG;
 
   if (ResetConnections (true))
-    IncrementUndoSerialNumber ();
+    {
+      IncrementUndoSerialNumber ();
+      Draw (); /* XXX: Not sure if this is required */
+    }
 
   User = false;
 
diff --git a/src/report.c b/src/report.c
index 3d74a26..7ec0370 100644
--- a/src/report.c
+++ b/src/report.c
@@ -46,6 +46,7 @@
 #include "macro.h"
 #include "undo.h"
 #include "find.h"
+#include "draw.h"
 
 #ifdef HAVE_LIBDMALLOC
 #include <dmalloc.h>
@@ -689,7 +690,10 @@ ReportAllNetLengths (int argc, char **argv, int x, int y)
 
     got_one:
       if (ResetConnections (true))
-        IncrementUndoSerialNumber ();
+	{
+	  IncrementUndoSerialNumber ();
+	  Draw ();
+	}
       length = XYtoNetLength (x, y, &found);
 
       gui->log("Net %s length %.*f %s\n", netname, prec, length*scale, units_name);
@@ -705,7 +709,10 @@ ReportNetLength (int argc, char **argv, int x, int y)
   int found = 0;
 
   if (ResetConnections (true))
-    IncrementUndoSerialNumber ();
+    {
+      IncrementUndoSerialNumber ();
+      Draw ();
+    }
   gui->get_coords ("Click on a connection", &x, &y);
 
   length = XYtoNetLength (x, y, &found);
diff --git a/src/set.c b/src/set.c
index c5d34eb..25267d8 100644
--- a/src/set.c
+++ b/src/set.c
@@ -293,7 +293,10 @@ SetMode (int Mode)
       if (Mode == LINE_MODE && TEST_FLAG (AUTODRCFLAG, PCB))
 	{
 	  if (ResetConnections (true))
-	    IncrementUndoSerialNumber ();
+	    {
+	      IncrementUndoSerialNumber ();
+	      Draw ();
+	    }
 	}
     }
